src/Polynomials.jl

```diff
@@ -415,29 +419,25 @@ polyder(Poly([1, 3, -1]))   # Poly(3 - 2x)
 ```
 """
 function polyder{T}(p::Poly{T}, order::Int=1)
-    n = length(p)
-    if order < 0
-        error("Order of derivative must be non-negative")
-    elseif n <= order
-        return Poly(zeros(T,0),p.var)
-    elseif order == 0
-        return p
-    else
-        a2 = Array(T, n-order)
-        for i = order:n-1
-            a2[i-order+1] = p[i] * prod((i-order+1):i)
-        end
-        return Poly(a2, p.var)
-    end
+  n = length(p)
+  order < 0         && error("Order of derivative must be non-negative")
+  order == 0        &&                    return p
+  T<:AbstractFloat  && any(isnan(p.a)) && return Poly(T[NaN], p.var)
+  n <= order        &&                    return Poly(T[], p.var)
+
+  a2 = Array(T, n-order)
+  for i = order:n-1
+    a2[i-order+1] = p[i] * prod((i-order+1):i)
+  end
+
+  return Poly(a2, p.var)
 end
 ctranspose{T}(p::Poly{T}) = polyder(p)
 
-polyint{T}(a::Array{Poly{T},1}, k::Number  = 0) = [ polyint(p,k) for p in a ]
-polyder{T}(a::Array{Poly{T},1}, order::Int = 1) = [ polyder(p,order) for p in a ]
 polyint{n,T}(a::Array{Poly{T},n}, k::Number  = 0) = map(p->polyint(p,k),a)
```

Comment:
I think we can remove lines like

```
#polyint{T}(a::Array{Poly{T},1}, k::Number  = 0) = [ polyint(p,k) for p in a ]
```

and replace with

```
@deprecate polyint{T}(a::Array{Poly{T},1}, k::Number  = 0) [ polyint(p,k) for p in a ]
```

But it would be nicer to suggest `polyint.(a,k)` when working with v"0.5" though, so maybe there is a more refined way.